### PR TITLE
Recently viewed products list fetched via ajax 

### PR DESCRIPTION
**Pull request by damianlegawiec**
1. This will allow us to incorporate full page caching without any breaking functionality to this extension
2. Updated development dependencies & fixes one rubocop issue

### CODE REVIEW

File: app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Spree.fetch_recently_viewed_products = (productId = "")->
+  $.ajax
```
Comment (priyank-gupta) [inline app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee]: @damianlegawiec I think we should only send request if `productID` is not blank.

File: app/controllers/spree/products_controller_decorator.rb [hunk 2]
```diff
@@ -1,9 +1,18 @@
 module Spree
   ProductsController.class_eval do
-    after_action :recently_viewed, only: :show
+    skip_before_action :set_current_order, only: :recently_viewed
+    after_action :save_recently_viewed, only: :recently_viewed
```
Comment (priyank-gupta) [inline app/controllers/spree/products_controller_decorator.rb]: @damianlegawiec I think this callback should be after `#show` and not `#recently_viewed`.

File: app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    url: Spree.pathFor("recently_viewed_products?product_id=" + productId),
```
Comment (priyank-gupta) [inline app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee]: @damianlegawiec Not sure why are we sending productID here if we plan to fetch all recently viewed products.
I think I am missing something here.

Comment (damianlegawiec) [inline app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee]: @priyank-gupta we're sending product id to save it also in recently viewed list - that way it is all done in one single request

BTW I see that right now specs are failing - will fix it

File: app/controllers/spree/products_controller_decorator.rb (hunk 2 above)
Comment (damianlegawiec) [inline app/controllers/spree/products_controller_decorator.rb]: @priyank-gupta if show action will be cached it won't fire, that's why it's after recently_viewed

File: app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee (hunk 1 above)
Comment (damianlegawiec) [inline app/assets/javascripts/spree/frontend/add_recently_viewed_products.js.coffee]: @priyank-gupta we want to fetch this also on products list, taxon pages and so on, prouctId is needed only for adding a product to recently viewed list

**Comment by priyank-gupta** [comment]
@damianlegawiec Got it. Seems fine to me ( with specs passed :) ) :+1:

**Comment by damianlegawiec** [comment]
@priyank-gupta specs fixed!

**Pull request merged**